src/dmd/backend/elfobj.d

```diff
@@ -1195,18 +1198,47 @@ void ElfObj_term(const(char)* objfilename)
             sechdr = &SecHdrTab[seg.SDrelidx];
             sechdr.sh_size = cast(uint)seg.SDrel.length();
             sechdr.sh_offset = foffset;
+
+            // sort the relocations by offset
             if (I64)
             {
                 assert(seg.SDrelcnt == seg.SDrel.length() / Elf64_Rela.sizeof);
-                debug for (size_t j = 0; j < seg.SDrelcnt; ++j)
-                {
-                    Elf64_Rela *p = (cast(Elf64_Rela *)seg.SDrel.buf) + j;
-                    if (ELF64_R_TYPE(p.r_info) == R_X86_64_64)
-                        assert(*cast(Elf64_Xword *)(seg.SDbuf.buf + p.r_offset) == 0);
-                }
+                extern (C) @trusted nothrow
+                    static int elf64_rel_fp(scope const(void*) e1,
+                                            scope const(void*) e2)
+                    {
+                        Elf64_Rela *r1 = cast(Elf64_Rela *)e1;
+                        Elf64_Rela *r2 = cast(Elf64_Rela *)e2;
+
+                        return (r1.r_offset > r2.r_offset) - (r1.r_offset < r2.r_offset); 
+                    }
+                qsort(
+                    seg.SDrel.buf,
+                    seg.SDrel.length() / Elf64_Rela.sizeof,
+                    Elf64_Rela.sizeof,
+                    &elf64_rel_fp
+                );
             }
             else
+            {
                 assert(seg.SDrelcnt == seg.SDrel.length() / Elf32_Rel.sizeof);
+                extern (C) @trusted nothrow
+                    static int elf32_rel_fp(scope const(void*) e1,
+                                            scope const(void*) e2)
+                    {
+                        Elf32_Rel *r1 = cast(Elf32_Rel *)e1;
+                        Elf32_Rel *r2 = cast(Elf32_Rel *)e2;
+
+                        return cast(int)(r1.r_offset - r2.r_offset);
```

Comment:
I meant here.

Reply:
32bit offset.
that works

Reply:
OK

Reply:
> 32bit offset. that works

not if the difference of the `uint`s is larger than `1<<31`

Reply:
I changed the code already before it was merged.